src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -314,8 +352,32 @@ def csv(self, path: str) -> DataFrame:
         Returns:
             a :class:`DataFrame` that is set up to load data from the specified CSV file(s) in a Snowflake stage.
         """
+        # infer schema is set to false by default
+        if "INFER_SCHEMA" not in self._cur_options:
+            self._cur_options["INFER_SCHEMA"] = False
+        self._infer_schema = self._cur_options["INFER_SCHEMA"]
```

Comment:
Are both `self._infer_schema` and `self._cur_options["INFER_SCHEMA"]` immutable? If not then we should only keep one copy so that we don't need to maintain them to be in sync.

Reply:
I can work on that refactor